src/part2/interacting-with-contracts.md

```diff
@@ -0,0 +1,92 @@
+# 6. Interacting with Contracts
+
+## Effective Contract Interaction
+
+This section is dedicated to understanding how to interact with smart contracts in CosmWasm effectively. We'll cover managing contract state, executing operations, and the nuances of data interaction.
+
+## Managing Contract State
+
+Efficient state management is crucial for the smooth operation of smart contracts. This part discusses state management in CosmWasm contracts.
+
+### State Management Techniques
+
+- **State Storage Strategies**: Best practices for storing and retrieving state within your contracts.
+- **State Migration**: How to handle state migration during contract upgrades.
+
+```rust
+#[cw_serde]
+struct Config {
+    owner: Addr,
+    amount: Uint128,
+    // using a struct for more complex state allows you to add more fields later on without breaking the contract
+}
+
+const CONFIG: Item<Config> = Item::new("config");
+
+/// Maps an address to a balance
+/// You can use this to store a balance per address and load it later
+const BALANCE: Map<Addr, Uint128> = Map::new("balance");
+```
+
+### Sending and Querying Data
+
+Interacting with contracts involves sending transactions and querying data. This section provides detailed guidance on these operations.
+
+### Sending Transactions
```

Comment:
Is this section about interacting from outside of a contract or sending messages to another contract from inside a contract?
- If it's about interacting from the outside: I don't know if there is a Rust framework for doing that. There's cosmjs for doing it in Typescript or you can use the chain binary.
- If it's about contract to contract messages: Those are not called "transactions". They are called "messages". A transaction can contain multiple messages. Inside the contract, you can add messages to the transaction, but not a new transaction.

I added some example code for contract to contract interactions here.

Reply:
Thank you, @chipshort, for your feedback. I've updated the documentation to clarify external versus internal contract interactions and corrected the terminology to "messages" for contract-to-contract communication. Here's a summary:

External Interactions: Explained how to interact with contracts using CosmJS or CLI tools.
Contract-to-Contract Communications: Revised to highlight message-passing within CosmWasm.